### ISSUE TITLE
Hide non-public Sites Resource Models

### ISSUE THREAD

**Issue by bferguso** [bcgov/BCHeritage#694]
- Restrict sites visible on map (override mvt?) (This follows from the resource model permissions so it should do the right thing
- Heritage Site search results need to be limited to public sites
- Remove all access to Site Submission, Government Person, Sandcastle, Heritage Site Historical Date resource models
- Okay to show Heritage Site, Legislative Act, Government Resource Models

We will need to 

- [x] Write a function for each resource model that sets the permission on save
- [x] Set the permissions of all the private resource models that have no public access
- [x] Remove resource type filters on the search bar

Note that it appears that the search suggestions do not adhere to the resource permissions (the display values, etc. will still be displayed event though the resource model won't be found).

**Comment by bferguso** [comment]
This should be able to be achieved by way of resource model functions. The current default implementation is by way of "user_has_no_x_access" (read, edit, delete). Can create a function for each resource model that sets the permissions based on some attribute data.

There is also a manual way to set these permissions that we can consider if it is more desirable. However this would be independent of other attributes.

This should be a reasonable way to manage permissions because the majority of the records are publicly visible.

**Comment by bferguso** [comment]
Site criteria defined in #353

**Comment by emjohnst** [comment]
@bferguso During meeting today site DkSr-12 is appearing in search results (in delivery) despite its official recognition status and Registered type Cancelled.

**Comment by bferguso** [comment]
During today's meeting @emjohnst noticed that there were restricted sites visible to the public. After reviewing, this was due to those sites not having any Official Recognition Information. This has been corrected in the migration, and will be redeployed to DLVR and then TEST once it's confirmed to be correct.

I have also confirmed that saving existing or new resources without that data does behave properly, and is not displayed to the public users.
